src/Darwin/OSLog.cs

```diff
@@ -0,0 +1,121 @@
+//
+// OSLog.cs: Bindings to the os_log(3) API from Apple
+//
+// Authors:
+//  William Kent
+//
+// Copyright 2019, Xamarin Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using ObjCRuntime;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+
+namespace Darwin {
```

Comment:
I'm wondering if `ObjCRuntime` wouldn't be a better namespace for this. @spouliot what do you think?

Reply:
hmm... we have the reflex of using `ObjCRuntime` as both `ObjC` support and `Runtime` service types :) but

* The Mac-only `SystemLog` type is already in `Darwin`;
* the `Darwin` namespace is not part of Xamarin.iOS/tvOS/watchOS

so I don't have really hard feelings on either location. I would have used `ObjCRuntime` myself but, like mentioned, more a reflex that a thoughtful decision :)